cql3/expr/expression.cc

```diff
@@ -793,7 +775,11 @@ static value_set possible_lhs_values(const column_definition* cdef,
                             }
                             return unbounded_value_set;
                         },
-                        [&] (token) -> value_set {
+                        [&] (const function_call& token_fun_call) -> value_set {
```

Comment:
nitpick: at this point you don't know that token_fun_call is a token function call (you'll check this next), so maybe better to call this variable just fun_call?

Reply:
I wanted to indicate that the only acceptable `function_call` here is a call to the `token()` function. Any other `function_call` that happens to be there means that there's a bug and it will trigger an `on_internall_error`.
In that sense I know that this is a call to the token() function - anything else would mean that there's a bug.